src/System.Linq.Expressions/src/System/Linq/Expressions/Common/ArrayBuilderExtensions.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Runtime.CompilerServices;
+
+namespace System.Linq.Expressions
+{
+    /// <summary>
+    /// Helper extension methods for <see cref="ArrayBuilder{T}"/>.
+    /// </summary>
+    internal static class ArrayBuilderExtensions
+    {
+        /// <summary>
+        /// Returns a read-only collection wrapper around the array that was built.
+        /// </summary>
+        public static ReadOnlyCollection<T> ToReadOnly<T>(this ArrayBuilder<T> builder)
+        {
+            return new TrueReadOnlyCollection<T>(builder.ToArray());
```

Comment:
How many places is this method called?  I'm wondering if we really need this whole extension class or if we could just use ```new TrueReadOnlyCollection<T>(builder.ToArray())``` at each call site rather than ```builder.ToReadOnly()```.